### PR TITLE
[refactor] refactor TransactionEnvelope as Envelope

### PR DESCRIPTION
**Pull request by punwai**
Changes TransactionEnvelope type to be based on the Envelop generic type

### CODE REVIEW

File: crates/sui-types/src/messages.rs [hunk 1]
```diff
@@ -496,98 +497,56 @@ impl TransactionData {
...
             return Ok(());
         }
-        self.signed_data
-            .tx_signature
-            .verify(&self.signed_data.data, self.signed_data.data.sender)
+        self.tx_signature.verify(&self.data, self.data.sender)
+    }
+
+    fn add_to_verification_obligation(&self, _: &mut VerificationObligation) -> SuiResult<()> {
+        Ok(())
```
[the remark this answers is not in the document]
Comment (punwai) [reply crates/sui-types/src/messages.rs:522]: I think I'll return an error here.

[the remark this answers is not in the document]
Comment (lxfind) [reply crates/sui-types/src/messages.rs:522]: hmm are you saying that we will soon get rid of all the verification_obligation stuff? i.e. no way to do batch verification?
I don't have much context on that, but this PR shouldn't change the semantics. If we will get rid of the verification_obligation, we can do so in that switch.

[the remark this answers is not in the document]
Comment (punwai) [reply crates/sui-types/src/messages.rs:522]: Sure, I'll add it in this PR, and I'll delete this when we switch to BLS.

**Review by randall-Mysten — APPROVED** [review]

File: crates/sui-types/src/messages.rs [hunk 2]
```diff
@@ -496,98 +497,59 @@ impl TransactionData {
...
-            .tx_signature
-            .verify(&self.signed_data.data, self.signed_data.data.sender)
+        self.tx_signature.verify(&self.data, self.data.sender)
     }
 
+    fn add_to_verification_obligation(
+        &self,
+        _: &mut VerificationObligation,
+    ) -> SuiResult<()> {
+        panic!("CRITICAL: SenderSignedData::add_to_verification_obligation is called. This function is yet to be implemented and should never be called.")
```
Comment (punwai) [inline crates/sui-types/src/messages.rs:525]: @lxfind I added this instead - this function should never be called. Is this alright?

Comment (punwai) [reply crates/sui-types/src/messages.rs:527]: Also @lxfind ready for review.

**Review by lxfind — APPROVED** [review]

**Comment by randall-Mysten** [comment]
closing stale PRs, please reopen if still relevant. Sorry we forgot to merge this for you.

**Pull request closed without merging**